### PR TITLE
Convert JS tools to ESM

### PR DESCRIPTION
**Pull request by nwalters512**
This was easy enough to do in isolation ahead of the full ESM cutover.

### CODE REVIEW

File: tools/rewrite-migrations.js
Comment (nwalters512) [inline tools/rewrite-migrations.js:1]: This script no longer does anything useful for us.

File: tools/dump-anonymize.mjs [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
+// $ node dump-anonymize.mjs <data-dump-folder> <anonymous-output-folder>
 
-const fs = require('fs');
-const async = require('async');
-const { v4: uuidv4 } = require('uuid');
-const path = require('path');
+import fs from 'node:fs';
+import async from 'async';
+import { v4 as uuidv4 } from 'uuid';
+import path from 'node:path';
```
Comment (jonatanschroeder) [inline tools/dump-anonymize.mjs:7]:
```suggestion
import * as fs from 'node:fs';
import * as async from 'async';
import { v4 as uuidv4 } from 'uuid';
import * as path from 'node:path';
```

Same in other changes.

Comment (nwalters512) [reply tools/dump-anonymize.mjs:7]: Why? We want to be moving the other direction (away from `import * as ...`) as we move to native ESM.

Comment (jonatanschroeder) [reply tools/dump-anonymize.mjs:7]: Ok, I was unaware this was possible in ESM. Carry on...

Comment (nwalters512) [reply tools/dump-anonymize.mjs:7]: Node does some interesting stuff to make this sorta Just Work (https://nodejs.org/api/esm.html#commonjs-namespaces):

> For better compatibility with existing usage in the JS ecosystem, Node.js in addition attempts to determine the CommonJS named exports of every imported CommonJS module to provide them as separate ES module exports using a static analysis process.

See also https://github.com/nodejs/cjs-module-lexer.

**Review by mwest1066 — APPROVED** [review]
Looks good to me, although I'd also be happy with a direct cutover to `.ts` if that's possible with your recent work.

**Comment by nwalters512** [comment]
@mwest1066 I'm deliberately not converting things in `tools/` to TS (yet) since it's nice to be able to execute them directly with `node ...` instead of `yarn tsx ...`. I'm open to changing my stance on this eventually.

**Pull request merged**